src/worker.cc

```diff
@@ -983,13 +984,21 @@ void from_json(rapidjson::Document& doc, Options::Action action, Api& api) {
       rapidjson::get_child_optional(doc, doc.HasMember("avoid_polygons") ? "/avoid_polygons"
                                                                          : "/exclude_polygons");
   if (rings_req) {
-    auto* rings_pbf = options.mutable_exclude_polygons();
-    try {
-      for (const auto& req_poly : rings_req->GetArray()) {
-        auto* ring = rings_pbf->Add();
-        parse_ring(ring, req_poly);
-      }
-    } catch (...) { throw valhalla_exception_t{137}; }
+    if (!rings_req->IsArray()) {
+      add_warning(api, 204);
+    } else {
+      auto* rings_pbf = options.mutable_exclude_polygons();
+      try {
+        for (const auto& req_poly : rings_req->GetArray()) {
+          if (!req_poly.IsArray() || (req_poly.IsArray() && req_poly.GetArray().Empty())) {
+            add_warning(api, 204);
+            continue;
+          }
+          auto* ring = rings_pbf->Add();
+          parse_ring(ring, req_poly);
+        }
+      } catch (...) { throw valhalla_exception_t{137}; }
+    }
```

Comment:
now we're checking if the outer and inner arrays are indeed arrays and that the inner arrays have a size. if not, we're adding a warning